bears/general/InvalidLinkBear.py

```diff
@@ -42,7 +42,7 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, ignore_regex):
         ignore_regex = re.compile(ignore_regex)
         regex = re.compile(
-            r'((ftp|http)s?://\S+\.(?:[^\s()\'\"<>|]+|'
+            r'((ftp|http)s?://[^\.:\s_/?#\[\]@]+\.(?:[^\s()\'\"<>|]+|'
```

Comment:
every metacharacter excecpt `[` and `]` loses its meaning, so you don't need to escape the stuff (except the brackets)